src/uu/tail/src/tail.rs

```diff
@@ -356,49 +636,543 @@ pub fn uu_app<'a>() -> Command<'a> {
+                if let Some(event_path) = paths.first() {
+                    if files.map.contains_key(event_path) {
+                        let _ = watcher.unwatch(event_path);
+                    }
+                }
+            }
+            Ok(Err(notify::Error {
+                kind: notify::ErrorKind::MaxFilesWatch,
+                ..
+            })) => crash!(1, "{} resources exhausted", text::BACKEND),
+            Ok(Err(e)) => crash!(1, "{:?}", e),
+            Err(mpsc::RecvTimeoutError::Timeout) => {
+                _timeout_counter += 1;
+            }
+            Err(e) => crash!(1, "RecvError: {:?}", e),
+        }
+
+        // main print loop
+        for path in files.map.keys().cloned().collect::<Vec<_>>() {
+            read_some = files.tail_file(&path, settings.verbose)?;
+        }
+
+        if !read_some && settings.pid != 0 && process.is_dead() {
+            // pid is dead
+            break;
+        }
+
+        if _timeout_counter == settings.max_unchanged_stats {
+            // TODO: [2021-10; jhscheer] implement timeout_counter for each file.
+            // ‘--max-unchanged-stats=n’
+            // When tailing a file by name, if there have been n (default n=5) consecutive iterations
+            // for which the file has not changed, then open/fstat the file to determine if that file
+            // name is still associated with the same device/inode-number pair as before. When
+            // following a log file that is rotated, this is approximately the number of seconds
+            // between when tail prints the last pre-rotation lines and when it prints the lines that
+            // have accumulated in the new log file. This option is meaningful only when polling
+            // (i.e., without inotify) and when following by name.
+            // TODO: [2021-10; jhscheer] `--sleep-interval=N`: implement: if `--pid=p`,
+            // tail checks whether process p is alive at least every N seconds
+        }
+    }
+    Ok(())
+}
+
+fn handle_event(
+    event: &notify::Event,
+    files: &mut FileHandling,
+    settings: &Settings,
+    watcher: &mut Box<dyn Watcher>,
+    orphans: &mut Vec<PathBuf>,
+) -> UResult<()> {
+    use notify::event::*;
+
+    if let Some(event_path) = event.paths.first() {
+        if files.map.contains_key(event_path) {
+            let display_name = files
+                .map
+                .get(event_path)
+                .unwrap()
+                .display_name
+                .to_path_buf();
+
+            match event.kind {
+                EventKind::Modify(ModifyKind::Metadata(MetadataKind::Any))
+                // | EventKind::Access(AccessKind::Close(AccessMode::Write))
+                | EventKind::Modify(ModifyKind::Metadata(MetadataKind::WriteTime))
+                | EventKind::Create(CreateKind::File)
+                | EventKind::Create(CreateKind::Folder)
+                | EventKind::Create(CreateKind::Any)
```

Comment:
Might be a nice use-case for the nested or patterns:
```
| EventKind::Create(CreateKind::File | CreateKind::Folder | CreateKind::Any)
```

Reply:
Yes, that's much better.